tasks/windows_client.yml

```diff
@@ -0,0 +1,106 @@
+---
+# tasks/windows_client.yml: Deploy various client-side configurations for Sensu-client on Windows
+# This contains all tasks, but these should be moved off into their respective places.
+
+    #- include_vars: "{{ ansible_distribution }}.yml"
+  - include_vars: "{{ ansible_os_family }}.yml"
+
+    # win_package throws an SSL error with a remote source. Using two steps here as a workaround
+  - name: Download the Sensu installer msi into a local dir
+    win_get_url:
+      url: "{{ sensu_msi_download_path }}"
+      dest: "c:\\temp\\sensu-installer.msi"
+      skip_certificate_validation: true
+
+    # If a version already exists on the target system, it should be made absent first.
+  - name: Install sensu-client from local dir
+    win_package:
+      path: "c:\\temp\\sensu-installer.msi"
+      product_id: "{{ sensu_msi_product_id }}"
+      arguments: /qn
+      force_extension: msi
+      validate_certs: no
+      state: present
+
+  - name: Build the config folders
+    win_file:
+      dest: "{{ sensu_win_base_path }}\\conf.d\\ssl"
+      state: directory
+
+  - name: Deploy Sensu Redis configuration
+    win_template:
+      dest: "{{ sensu_win_base_path }}\\conf.d\\redis.json"
+      src: "{{ sensu_redis_config }}"
+    when: sensu_deploy_redis
+    notify:
+      - restart sensu-client service
+
+  - name: Deploy Sensu RabbitMQ configuration on Windows
+    win_template:
+      dest: "{{ sensu_win_base_path }}\\conf.d\\rabbitmq.json"
+      src: "{{ sensu_win_rabbitmq_config }}"
+    when: sensu_transport == "rabbitmq"
+    notify:
+      - restart sensu-api service
+      - restart sensu-client service
+
+  - name: Deploy Sensu transport configuration on Windows
+    win_template:
+      dest: "{{ sensu_win_base_path }}\\conf.d\\transport.json"
+      src: transport.json.j2
+    notify:
+      - restart sensu-client service
+
+  - name: Deploy Sensu client service configuration on Windows
+    win_template:
+      dest: "{{ sensu_win_base_path }}\\conf.d\\client.json"
+      src: "{{ sensu_client_config  }}"
+    notify: restart sensu-client service
+
+  - name: Deploy the Sensu client SSL cert/key on Windows
+    win_copy:
+      src: "{{ item.src }}"
+      remote_src: "{{ sensu_ssl_deploy_remote_src }}"
+      dest: "{{ sensu_win_base_path }}\\conf.d\\ssl\\{{ item.dest }}"
+    with_items:
+      - { src: "{{ sensu_ssl_client_cert }}", dest: cert.pem }
+      - { src: "{{ sensu_ssl_client_key }}", dest: key.pem }
+    notify: restart sensu-client service
+    when: sensu_ssl_gen_certs
+
+  - name: Install the sensu-client service
+    win_service:
+      name: sensu-client
+      path: 'c:\opt\sensu\bin\sensu-client.exe'
+      display_name: Sensu Client
+      description: The Sensu Client Service
+      start_mode: delayed
+
+  - name: Fetch sensu-plugins-windows repo 
```

Comment:
Are we unable to re-use the existing logic here, https://github.com/sensu/sensu-ansible/blob/master/tasks/plugins.yml#L36-L46 to deploy plugins to Windows hosts?

Reply:
We probably can... what if we are using a mixed windows and Linux environments and we don't want to mix the checks together... i'll have a think about it and will need to test

Reply:
hmmm great question, I'm not experienced enough with Sensu on Windows to make a solid call on that side. Do some of the common plugins like https://github.com/sensu-plugins/sensu-plugins-cpu-checks not work? If I had to hazard a guess, some of them might just work whereas I'd expect there be specific ones that might bomb out completely.

Reply:
I know there are a handful that will fail I recall at least one (can't recall which one off the top of my head) that relies on `linux-kstat`. Basically if you have a mixed environment you need to install plugins based on subscription -> plugin mapping.

Reply:
The sensu for windows installer graciously installs a bunch of useful linux utils (grep, awk, etc) so it's very possible that some linux checks that depend on these tools would work. My experience so far however is purely with deploying the powershell based win checks.